jevents/resolve.c

```diff
@@ -544,7 +544,7 @@ int jevent_name_to_attr_extra(const char *str, struct perf_event_attr *attr,
 			char *gs;
 			int ret;
 
-			asprintf(&gs, "/sys/devices/uncore_%s_*", pmu);
+			asprintf(&gs, "/sys/devices/uncore_%s_[0-9]", pmu);
```

Comment:
In glob the pattern `[0-9]` matches only a single digit. So it will omit PMU number 10 and so on.
The better choice might be to use the pattern: `/sys/devices/uncore_%s_[0-9]*`.

Reply:
Good catch; I didn't think about that.

Unfortunately, though, `glob`'s pattern-matching leaves a bit to be desired: the pattern `[0-9]*` won't just match multiple digits, it'll match one digit and any characters thereafter.  That's certainly closer to what we want, and I can't think of any scenarios in which it would fail, but it still seems like we need something a little more robust.

What we really want here is the regular expression `[0-9]+`, but I'm not sure I want to introduce an additional dependency on `regex.h` to solve this simple bug.

Back to `glob`: do you know of any way to match at least one digit?

Reply:
You could do an open coded check. Or regex is fine too since it's part of the standard library